app/scripts/directives/oscSecrets.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .directive("oscSecrets", function($uibModal, $timeout) {
+    return {
+      restrict: 'E',
+      scope: {
+        pickedSecrets: "=model",
+        allSecrets: '=',
+        namespace: "=",
+        additionalInputFieldName: "@?",
+        additionalInputFieldTooltip: "@?",
+        type: "@",
+        alertsCallback: '&',
+        loadSecretsCallback: '&'
```

Comment:
I don't think we need `loadSecretsCallback`. Can we just use `$scope.$watch('allSecrets')` in the controller that uses this directive?